api/src/main/java/org/commonjava/maven/galley/model/Transfer.java

```diff
@@ -461,17 +461,13 @@ public Transfer getSiblingMeta( final String extension )
             return null;
         }
 
-        final String named = resource.getPath() + extension;
+        final String generatedPath = provider.getGeneratedPath( resource );
```

Comment:
I think rather than doing this, we could construct the Transfer instance with the ConcreteResource, which contains the logical path, and the calculated storage path (separately). Then, we should add a new method to Transfer called `getStoragePath()` which uses the calculated path.

When we do a getSiblingMeta() call, we can append the extension to each path and construct the new Transfer accordingly.

Reply:
Will do. And seems that we should also change almost all transfer.getPath() call to this transfer.getStoragePath().